### PR TITLE
Add workflow to publish image

### PR DESCRIPTION
**Pull request by calvwang9**
Setting up CICD to deploy timelock-worker on the production testnet env

guide ref: https://www.notion.so/chainlink/Create-CI-Pipeline-to-Build-Push-Docker-image-9c1306e0f2c940e39d581b0968068018

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    environment: publish
+    permissions:
+      id-token: write
+      contents: write
+      actions: read
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Build and push image
+        uses: ./.github/actions/build-push
```
Comment (0xAustinWang) [inline .github/workflows/publish.yml]: where is this defined?

Comment (calvwang9) [reply .github/workflows/publish.yml]: copy-paste error, thanks :)

**Comment by fbac** [comment]
Hey Calvin, is this workflow going to publish public or private images?

**Comment by calvwang9** [comment]
@fbac will be publishing to a private ECR repo, following this guide https://www.notion.so/chainlink/Mapping-End-to-end-App-Deployment-e92e381bddff4cb2a78191f15931b144

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        uses: actions/checkout@v4
+
+      - name: Build and push image
+        uses: smartcontractkit/.github/actions/cicd-build-publish-artifacts-go@25645c21796ebb5554693fcc0d312dc88330fbe0 # cicd-build-publish-artifacts-go@0.2.0
+        with:
+          # general inputs
+          app-name: timelock-worker
+          publish: true
+          # aws inputs
+          aws-region: ${{ secrets.AWS_REGION }}
```
Comment (fbac) [inline .github/workflows/publish.yml:32]: Are those secrets already created?

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    permissions:
+      id-token: write
+      contents: write
+      actions: read
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Build and push image
+        uses: smartcontractkit/.github/actions/cicd-build-publish-artifacts-go@25645c21796ebb5554693fcc0d312dc88330fbe0 # cicd-build-publish-artifacts-go@0.2.0
```
Comment (fbac) [inline .github/workflows/publish.yml]: Is this github action public? It seems I can't find it.

Comment (calvwang9) [reply .github/workflows/publish.yml]: thanks, was a private action so replaced with a manual workflow

**Review by fbac — APPROVED** [review]
LGTM

**Comment by fbac** [comment]
@calvwang9 The workflow looks good to me. 
Only one nitpick: it's not possible to manually trigger a image publish (for example being able to trigger on workflow_dispatch) so you'd have to wait until the next tag. I guess you already accounted for this?

**Pull request merged**